scripts/fires/wfigs_data.py

```diff
@@ -1,68 +1,90 @@
+PRE_2022_FIRE_LOCATIONS_URL = "https://services3.arcgis.com/T4QMspbfLg3qTGWY/arcgis/rest/services/Fire_History_Locations_Public/FeatureServer/0/query?where=1%3D1&outFields=InitialLatitude,InitialLongitude,InitialResponseAcres,InitialResponseDateTime,UniqueFireIdentifier,IncidentName,IncidentTypeCategory,IrwinID,FireCauseSpecific,FireCauseGeneral,FireCause,FireDiscoveryDateTime,ContainmentDateTime,ControlDateTime,IsCpxChild,CpxID,DiscoveryAcres,DailyAcres,POOFips,POOState,EstimatedCostToDate,TotalIncidentPersonnel,UniqueFireIdentifier&outSR=4326&orderByFields=FireDiscoveryDateTime&f=json&resultType=standard"
 _OUTPUT = "/cns/jv-d/home/datcom/v3_resolved_mcf/fire/wfigs/"
+_CACHE = {}
+
+with open(os.path.join(_SCRIPT_PATH, 'location_file.json')) as f:
+    data = f.read()
+
+_CACHE = json.loads(data)
 
 _FIRE_INCIDENT_MAP = {
     'CX': 'FireIncidentComplexEvent',
     'WF': 'WildlandFireEvent',
     'RX': 'PrescribedFireEvent'
 }
 
-_LOCATION_MAP = "\nlocation: {location}"
-_DAILY_ACRES_MAP = "[Acre {daily_acres}]"
-_INITIAL_RESPONSE_AREA_MAP = "[Acre {initial_response_area}]"
-_EXPECTED_LOSS_MAP = "[USDollar {costs}]"
+_DAILY_ACRES_MAP = "Acre {daily_acres}"
+_INITIAL_RESPONSE_AREA_MAP = "Acre {initial_response_area}"
+_EXPECTED_LOSS_MAP = "USDollar {costs}"
 
 
 def get_data(url):
     """Get data from the API.
 
-  This method does that by retrieving the max results in a loop till we get to
-  the end of the dataset.
+    This method does that by retrieving 32K records(API call return size) in a
+    loop till we get to the end of the dataset.
 
-  Args:
-    url: URL for the API call.
+    Args:
+      url: URL for the API call.
 
-  Returns:
-    A Pandas dataframe containing merged historical and YTD data.
-  """
-    r = requests.get(url)
-    x = r.json()
+    Returns:
+      A Pandas dataframe containing merged historical and YTD data.
+    """
     data = []
-    for row in x["features"]:
-        data.append(row["attributes"])
-    max_record_count = len(x["features"])
-    offset_str = "&resultOffset="
-    i = 1
-    while (x["features"]) and len(x["features"]) == max_record_count:
-        r = requests.get(url + offset_str + str(max_record_count * i))
-        x = r.json()
-        for row in x["features"]:
+    full_page_record_count = 0
+    i = 0
+    while True:
+        r = requests.get("{url}&resultOffset={record_count}".format(
+            url=url, record_count=str(full_page_record_count * i)))
+        response_json = r.json()
+        for row in response_json["features"]:
             data.append(row["attributes"])
         i += 1
+        full_page_record_count = max(full_page_record_count,
+                                     len(response_json["features"]))
```

Comment:
did you encounter inconsistent page sizes or is this to accommodate the loop? either way, that's worth commenting here.